test/jsflags.spec.js

```diff
@@ -53,3 +53,20 @@ describe('canaryGetOptions', function () {
     ])
   })
 })
+
+describe('headlessGetOptions', function () {
+  var headlessGetOptions = launcher.test.headlessGetOptions
+
+  it('should return the headless flags', function () {
+    var parent = sinon.stub().returns(['-incognito'])
+    var context = {}
+    var url = 'http://localhost:9876'
+    var args = {}
+    expect(headlessGetOptions.call(context, url, args, parent)).to.be.eql([
+      '-incognito',
+      '--headless',
+      '--disable-gpu',
```

Comment:
`--disable-gpu` is currently needed with headless because of linking errors with osmesa.so. That will hopefully not be required once Chrome switches to their SwiftShader library as a software rendering engine. See https://bugs.chromium.org/p/chromium/issues/detail?id=617551 for details.

`-incognito` is just an example flag in the test, ensuring that it gets passed through.